### PR TITLE
Refactor some IAM policies

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request refactors some of the IAM policies in the primary Terraform configuration (in `terraform/`).

## 💭 Motivation and context ##

There were instances of unnecessary policies being created when a single, reused policy would work just as well. There were also some policies that were not set up for reuse when the policy either already saw use by multiple resources or it made sense to refactor it so that it could be reused.

## 🧪 Testing ##

Automated tests pass. I used these changes in my development environment with no issues.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: terraform/lambda_eni_policy.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#
+# These policy documents will be applied to the roles we are creating.
+data "aws_iam_policy_document" "lambda_eni_policy_doc" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface",
```
Comment (dav3r) [inline terraform/lambda_eni_policy.tf]: Alphabetize:
```suggestion
      "ec2:DeleteNetworkInterface",
      "ec2:DescribeNetworkInterfaces",
```

File: terraform/moe_bucket.tf [hunk 2]
```diff
@@ -8,3 +8,70 @@ resource "aws_s3_bucket" "moe_bucket" {
...
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.moe_bucket.arn}/*",
+    ]
```
Comment (dav3r) [inline terraform/moe_bucket.tf:35]: I think we can safely consolidate this to:
```suggestion
    actions = [
      "s3:GetObject",
      "s3:ListBucket",
    ]

    resources = [
      aws_s3_bucket.moe_bucket.arn,
      "${aws_s3_bucket.moe_bucket.arn}/*",
    ]
```

File: terraform/moe_bucket.tf [hunk 3]
```diff
@@ -8,3 +8,70 @@ resource "aws_s3_bucket" "moe_bucket" {
...
+    effect = "Allow"
+
+    actions = [
+      "s3:DeleteObject",
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.moe_bucket.arn}/*",
+    ]
```
Comment (dav3r) [inline terraform/moe_bucket.tf:69]: Similar consolidation:
```suggestion
      "s3:DeleteObject",
      "s3:ListBucket",
      "s3:PutObject",
    ]

    resources = [
      aws_s3_bucket.moe_bucket.arn,
      "${aws_s3_bucket.moe_bucket.arn}/*",
    ]
```

**Review by dav3r — APPROVED** [review]
LGTM, though I noted a few small items for improvement. 🚀

File: terraform/moe_bucket.tf (hunk 2 above)
Comment (jsf9k) [reply terraform/moe_bucket.tf:35]: I'm fine either way.

**Review by jsf9k — APPROVED** [review]
Looks good to me!

File: terraform/moe_bucket.tf (hunk 3 above)
Comment (jsf9k) [reply terraform/moe_bucket.tf:69]: I'm fine either way.  I can't decide which way I like better; both have their pluses and minuses.

File: terraform/moe_bucket.tf (hunk 2 above)
Comment (mcdonnnj) [reply terraform/moe_bucket.tf:35]: So this was based (I think) on the [AWS example IAM policy](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-policies-s3.html#iam-policy-ex0) and given some of the [more advanced examples](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-policies-s3.html#iam-policy-ex1) I'm wary of combining these permissions and having possibly unexpected interactions. Thoughts?

Comment (dav3r) [reply terraform/moe_bucket.tf:35]: I'm fine keeping it as is.  If it ain't broke....

File: terraform/moe_bucket.tf (hunk 3 above)
Comment (dav3r) [reply terraform/moe_bucket.tf:69]: See https://github.com/cisagov/cyhy_amis/pull/649#discussion_r1164469340

File: terraform/moe_bucket.tf (hunk 2 above)
Comment (mcdonnnj) [reply terraform/moe_bucket.tf:35]: Just for documentation I did test both IAM policy changes and expected functionality worked.

**Pull request merged**